netplan/cli/commands/ip.py

```diff
@@ -139,7 +140,10 @@ def lease_method_nm_connection():  # pragma: nocover (covered in autopkgtest)
 
         # Extract out of the generator our mapping in a dict.
         logging.debug('command ip leases: running %s', argv)
-        out = subprocess.check_output(argv, universal_newlines=True)
+        try:
+            out = subprocess.check_output(argv, universal_newlines=True)
+        except CalledProcessError as e:  # pragma: nocover (better be covered in autopkgtest)
+            sys.exit("Interface %s is not managed by netplan or not configured for DHCP" % (self.interface))
```

Comment:
Why print this message instead of returning an empty string?  The latter is far preferable for parsability; and certainly, we expect 'netplan ip leases' to be machine-consumable.

(It's ironic that currently we pass through the 'This is private data, do not parse' comment from netif/leases; we should really be filtering that, because regardless of whether systemd wants these treated as internal and not to be parsed, we are deliberately exposing them and need to deliver an interface guarantee here.)